File: MyApp.AngularClient/ClientApp/src/app/shared/models/layout.model.ts

import { NavItem } from "./navbar.model";

enum LayoutTemplates {
  Topbar = 0,
  Sidebar = 1
}
enum LayoutThemes {
  Default = 0
}

export class LayoutComponent {
  LayoutComponentID: number;

  LayoutTemplate: LayoutTemplates;
  LayoutTheme: LayoutThemes;
  NavItems: NavItem;

  /*
        public LayoutTemplates LayoutTemplate { get; set; }
        public LayoutThemes LayoutTheme { get; set; }

        public List<NavItemViewModel> NavItems { get; set; }

*/

}

File: MyApp.Model/Layout/LayoutModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyApp.Model.Layout
{

    public enum LayoutTemplates
    {
        Topbar,
        Sidebar
    }

    public enum LayoutThemes
    {
        Default
    }

    public class LayoutComponentModel
    {
        public int LayoutComponentID { get; set; }
    }

    public class LayoutComponentViewModel
    {
        public int LayoutComponentID { get; set; }

        public LayoutTemplates LayoutTemplate { get; set; }
        public LayoutThemes LayoutTheme { get; set; }
       
        public List<NavItemViewModel> NavItems { get; set; }
    }

}
